python/python/lancedb/embeddings/openai.py

```diff
@@ -79,6 +83,16 @@ def generate_embeddings(
     def _openai_client(self):
         openai = attempt_import_or_raise("openai")
 
-        if not os.environ.get("OPENAI_API_KEY"):
+        if self.api_key is None and not os.environ.get("OPENAI_API_KEY"):
```

Comment:
why not just let the API fail if missing?

Reply:
Its just a minor UX thing. To provide a better error message that tells user what to change exactly :)